### PR TITLE
JMESPath updates

### PR DESCRIPTION
**Pull request by chipzoller**
## Proposed Changes

Updates to the JMESPath docs (mostly)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [x] I have inspected the website preview for accuracy.
- [x] I have signed off my issue.

### CODE REVIEW

**Comment by chipzoller** [comment]
Folks, this is a huge update to the JMESPath page which covers ALL custom filters individually. I know it's a lot to review, but let me know what you think and what could be better, fixed, etc.

**Comment by chipzoller** [comment]
Maintainers, please review and either provide feedback or approve and squash/merge as this should be cherry picked to 1.6. I would like to merge this prior to sending further doc updates.

File: content/en/docs/Writing policies/jmespath.md [hunk 1]
```diff
@@ -434,51 +442,1271 @@ spec:
...
+* Duration (ex., '10h')
 
-### Pattern_match
+Note that how the inputs are enclosed determines how Kyverno interprets their type. Numbers enclosed in back ticks are scalar values while quantities and durations are enclosed in single quotes thus treating them as strings. Using the correct enclosing character is important because, in Kubernetes "regular" numbers are treated implicitly as units of measure. The number written \`10\` is interpreted as an integer or "the number ten" whereas '10' is interpreted as a string or "ten bytes". See the [Formatting](#formatting) section above for more details.
 
-### Regex_match
+| Input 1            | Input 2            | Output             |
+|--------------------|--------------------|--------------------|
+| Number             | Number             | Number             |
+| Quantity           | Quantity           | Quantity           |
```
Comment (samj1912) [inline content/en/docs/Writing policies/jmespath.md:761]: I would say that this isn't entirely a valid input set. Could we restrict the documentation to numbers for now. I would prefer to do more research on this before we document the behavior for quantities.

File: content/en/docs/Writing policies/jmespath.md [hunk 2]
```diff
@@ -434,51 +442,1271 @@ spec:
...
+* Number (ex., \`10\`)
+* Quantity (ex., '10Mi')
+* Duration (ex., '10h')
+
+Note that how the inputs are enclosed determines how Kyverno interprets their type. Numbers enclosed in back ticks are scalar values while quantities and durations are enclosed in single quotes thus treating them as strings. Using the correct enclosing character is important because, in Kubernetes "regular" numbers are treated implicitly as units of measure. The number written \`10\` is interpreted as an integer or "the number ten" whereas '10' is interpreted as a string or "ten bytes". See the [Formatting](#formatting) section above for more details.
+
+| Input 1            | Input 2            | Output   |
+|--------------------|--------------------|----------|
+| Number             | Number             | Number   |
+| Quantity           | Number             | Quantity |
```
Comment (samj1912) [inline content/en/docs/Writing policies/jmespath.md:1472]: I don't think this mathematically makes sense and I wonder if we should even document this.

File: content/en/docs/Writing policies/jmespath.md [hunk 3]
```diff
@@ -434,51 +442,1271 @@ spec:
...
+| Quantity           | Quantity           | Number   |
+| Duration           | Number             | Duration |
```
Comment (samj1912) [inline content/en/docs/Writing policies/jmespath.md:1474]: Same here.

**Review by samj1912 — APPROVED** [review]
Generally lgtm. I would prefer if we don't document some unsupported behaviors with arithmetic operators for now. Some of these I feel are invalid and we may not want to support these operations.

File: content/en/docs/Writing policies/jmespath.md (hunk 1 above)
Comment (chipzoller) [reply content/en/docs/Writing policies/jmespath.md:761]: Done. Will leave the note in place until we reach full resolution.

File: content/en/docs/Writing policies/jmespath.md (hunk 2 above)
Comment (chipzoller) [reply content/en/docs/Writing policies/jmespath.md:1472]: Will remove for now.

File: content/en/docs/Writing policies/jmespath.md (hunk 3 above)
Comment (chipzoller) [reply content/en/docs/Writing policies/jmespath.md:1474]: Will remove for now.

**Comment by chipzoller** [comment]
Comments addressed. Will now merge.

**Pull request merged**